test/uk/gov/hmrc/childcarecalculatorfrontend/controllers/ChildAgedTwoControllerSpec.scala

```diff
@@ -0,0 +1,173 @@
+
+    "be available" when {
+
+      "GET request is made" in {
+        val req = FakeRequest(GET, childAgedTwoPath).withSession(validSession)
+        val result = route(app, req)
+
+        result.isDefined shouldBe true
+        status(result.get) should not be NOT_FOUND
+      }
+    }
+  }
+
+  "ChildAgedTwoController" when {
+
+    "onPageLoad is called" should {
+
+      "load template successfully if there is no data in keystore" in {
+        when(
+          sut.keystore.fetchEntryForSession[Boolean](anyString)(any[HeaderCarrier], any[Reads[Boolean]])
+        ).thenReturn(
+          Future.successful(None)
+        )
+        val result = await(sut.onPageLoad(request.withSession(validSession)))
+        status(result) shouldBe OK
+        result.body.contentType.get shouldBe "text/html; charset=utf-8"
+      }
+
+      "load template successfully if there is data in keystore" in {
+        when(
+          sut.keystore.fetchEntryForSession[Boolean](anyString)(any[HeaderCarrier], any[Reads[Boolean]])
+        ).thenReturn(
+          Future.successful(Some(true))
+        )
+        val result = await(sut.onPageLoad(request.withSession(validSession)))
+        status(result) shouldBe OK
+        result.body.contentType.get shouldBe "text/html; charset=utf-8"
+      }
+
+      "redirect to error page if can't connect with keystore" in {
+        when(
+          sut.keystore.fetchEntryForSession[Boolean](anyString)(any[HeaderCarrier], any[Reads[Boolean]])
+        ).thenReturn(
+          Future.failed(new RuntimeException)
+        )
+        val result = await(sut.onPageLoad(request.withSession(validSession)))
+        status(result) shouldBe SEE_OTHER
+        result.header.headers("Location") shouldBe technicalDifficultiesPath
+      }
+    }
+
+    "onSubmit is called" when {
+
+      "there are errors" should {
+        "load same template and return BAD_REQUEST" in {
+          val form = new ChildAgedTwoForm(applicationMessagesApi).form.bind(
+            Map(
+              childAgedTwoKey -> ""
+            )
+          )
+
+          val result = await(
+            sut.onSubmit(
+              request
+                .withFormUrlEncodedBody(form.data.toSeq: _*)
+                .withSession(validSession)
+            )
+          )
+          status(result) shouldBe BAD_REQUEST
+          result.body.contentType.get shouldBe "text/html; charset=utf-8"
+        }
+      }
+
+      "saving in keystore is successful" in {
+        val form = new ChildAgedTwoForm(applicationMessagesApi).form.bind(
+          Map(
+            childAgedTwoKey -> "true"
+          )
+        )
+
+        when(
+          sut.keystore.cacheEntryForSession[Boolean](anyString, anyBoolean)(any[HeaderCarrier], any[Format[Boolean]])
+        ).thenReturn(
+          Future.successful(Some(true))
+        )
+
+        val result = await(
+          sut.onSubmit(
+            request
+              .withFormUrlEncodedBody(form.data.toSeq: _*)
```

Comment:
val result = await(
          sut.onSubmit(
            request
              .withFormUrlEncodedBody(childAgedTwoKey -> "true")
              .withSession(validSession)
          )
        )

strange, it does work for me you can refer branch CCALC-3008 as well for ExpectChildcareCostsControllerSpec